### DISCUSSION TITLE
How can I access yaml maps using Typescript?

### DISCUSSION THREAD

**Discussion by golergka** [eemeli/yaml#381] [Q&A]
I'm using version 2.0.1. I know that this document should be a map, and I want to access it's property named "key". I tried doing it two different ways: both using primitive values and creating ndoes myself, [as described in documentation](https://eemeli.org/yaml/#collections), but both result in errors:

```
declare const doc: yaml.Document.Parsed;

if (yaml.isMap(doc.contents)) {

  // Argument of type 'string' is not assignable to parameter of type 'ParsedNode'
  doc.contents.get("key")

  // Argument of type 'Node' is not assignable to parameter of type 'ParsedNode'.
  doc.contents.get(doc.createNode("key"))
}
```

How am I supposed to access this map?

**Comment by eemeli** [comment]
Yeah, the types are too tight here for access methods; will fix in a patch release. In the meantime, the same methods on the Document itself are more appropriately typed:
```js
declare const doc: yaml.Document.Parsed;
doc.get("key")
doc.get(doc.createNode("key"))
```
